server/datastore/mysql/apple_mdm.go

```diff
@@ -930,40 +931,227 @@ VALUES
+	var countArgs int
+	if len(hostIDs) > 0 {
+		countArgs++
+	}
+	if len(teamIDs) > 0 {
+		countArgs++
+	}
+	if len(profileIDs) > 0 {
+		countArgs++
+	}
+	if countArgs > 1 {
+		return errors.New("only one of hostIDs, teamIDs or profileIDs can be provided")
+	}
+	if countArgs == 0 {
+		return nil
+	}
+
+	var (
+		uuids    []string
+		args     []any
+		uuidStmt string
+	)
+
+	switch {
+	case len(hostIDs) > 0:
+		uuidStmt = `SELECT uuid FROM hosts WHERE id IN (?)`
+		args = append(args, hostIDs)
+
+	case len(teamIDs) > 0:
+		uuidStmt = `SELECT uuid FROM hosts WHERE `
+		if len(teamIDs) == 1 && teamIDs[0] == 0 {
+			uuidStmt += `team_id IS NULL`
+		} else {
+			uuidStmt += `team_id IN (?)`
+			args = append(args, teamIDs)
+			for _, tmID := range teamIDs {
+				if tmID == 0 {
+					uuidStmt += ` OR team_id IS NULL`
+					break
+				}
+			}
+		}
+
+	case len(profileIDs) > 0:
+		uuidStmt = `
+SELECT DISTINCT h.uuid
+FROM hosts h
+JOIN mdm_apple_configuration_profiles macp
+	ON h.team_id = macp.team_id OR (h.team_id IS NULL AND macp.team_id = 0)
+WHERE
+	macp.profile_id IN (?)`
+		args = append(args, profileIDs)
+	}
+
+	uuidStmt, args, err := sqlx.In(uuidStmt, args...)
+	if err != nil {
+		return ctxerr.Wrap(ctx, err, "prepare query to load host UUIDs")
+	}
+	if err := sqlx.SelectContext(ctx, ds.writer, &uuids, uuidStmt, args...); err != nil {
+		return ctxerr.Wrap(ctx, err, "execute query to load host UUIDs")
+	}
+
+	if len(uuids) == 0 {
+		return nil
+	}
+
+	const baseStmt = `
+INSERT INTO host_mdm_apple_profiles (
+	profile_id,
+	host_uuid,
+	profile_identifier,
+	profile_name,
+	operation_type,
+	status,
+	command_uuid
+)
+	-- NOTE: from https://dev.mysql.com/doc/refman/5.7/en/insert-on-duplicate.html
+	-- In an ON DUPLICATE KEY UPDATE clause:
+	-- References to columns from a UNION do not work reliably. To work
+	-- around this restriction, rewrite the UNION as a derived table so that its
+	-- rows can be treated as a single-table result set.
+	SELECT * FROM (
+		-- profiles to install, i.e. those part of the host's team/no team.
+		-- Except for the SELECT list, filtering on specific target IDs
+		-- (host/team/profile) and ignoring hmap rows that are already
+		-- "install" and NULL status, this is the same as
+		-- ListMDMAppleProfilesToInstall.
+		SELECT
+			ds.profile_id as profile_id,
+			ds.host_uuid as host_uuid,
+			ds.profile_identifier as profile_identifier,
+			ds.profile_name as profile_name,
+			? as operation_type,
+			NULL as status,
+			'' as command_uuid
+		FROM (
+			SELECT
+				macp.profile_id,
+				h.uuid as host_uuid,
+				macp.identifier as profile_identifier,
+				macp.name as profile_name
+			FROM mdm_apple_configuration_profiles macp
+				JOIN hosts h ON h.team_id = macp.team_id OR (h.team_id IS NULL AND macp.team_id = 0)
+				JOIN nano_enrollments ne ON ne.device_id = h.uuid
+			WHERE h.platform = 'darwin' AND ne.enabled = 1 AND ne.type = 'Device' AND h.uuid IN (?)
```

Comment:
Roberto, I added this condition based on what you found out yesterday about the `nano_enrollments` table (and the PR you created as a fix), the `ne.Type = 'Device'` part.